webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/policies/RuleTarget.scala

```diff
@@ -265,6 +266,76 @@ object RuleTarget extends Loggable {
     } }
   }
 
+  /**
+   * Return all node ids that match the set of target.
+   * allNodes pair is: (nodeid, isPolicyServer)
+   */
+  def getNodeIdsChunk(
+      targets : Set[RuleTarget]
+    , allNodes: Map[NodeId, Boolean /* isPolicyServer */]
+    , groups  : Map[NodeGroupId, Chunk[NodeId]]
+    , allNodesAreThere: Boolean = true  // if we are working on a subset of node, set to false
```

Comment:
this parameter does not seems to be used and it seems quite dangerous in the public method API (in rec method, it's OK, it's an implementation detail). Can you remove it ?

Reply:
it's the same as in here: https://github.com/Normation/rudder/pull/4290/files#diff-76a4f7665e1ec00753ea84eaab024d9eca6f32fed653555aba8cd9c70c14cb09L218
goal is to only use this method and phase out the getNodeIds that returns Set

Reply:
I don't get it. The linked methods does not have the `allNodesAreThere` parameter: 
`def getNodeIdsChunk(allGroups: Map[NodeGroupId, Chunk[NodeId]], targets: Set[RuleTarget], allNodeInfos: Map[NodeId, NodeInfo]) : Chunk[NodeId] = ...`

Reply:
yes, but i want to replace getNodeIds by getNodeIdsChunk in the future, so having the same signature ease that